xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -14875,6 +14880,10 @@ glusterd_check_brick_order(dict_t *dict, char *err_str, int32_t type,
     ai_list->info = NULL;
     CDS_INIT_LIST_HEAD(&ai_list->list);
 
+    pre_list = MALLOC(sizeof(addrinfo_list_t));
```

Comment:
nit: Shouldn't we check for MALLOC failure?
Also, we should `free` it at the end of the func, right?

Reply:
The MALLOC is actually a macro in gluster which internally calls a inline function `__gf_default_malloc` wherein the result of malloc operation is checked. If the memory allocation fails then the value of `pre_list` here would be NULL.

Reply:
Adding to that, I think we should have a check condition here, because if the MALLOC had failed then `pre_list` would be NULL, so technically any operation done on that would cause issues. Similar to what was done with `pre_list_tmp1`.

Reply:
That was the reason for not adding it earlier. But I agree to that doing a check will prevent if it failed

Reply:
Any specific reason to use MALLOC instead of using GF_MALLOC